config/azure/setBuildVars.py

```diff
@@ -52,25 +52,26 @@
 # trigger. These variables can be referenced in a pipeline yaml file downstream
 # of the step that runs this script.
 
-def getCommitHash() :
-	commit = os.environ.get( 'BUILD_SOURCEVERSION' )
-	# Azure merges the branch into its target in PR build, so
-	# BUILD_SOURCEVERSION isn't correct as it references the ephemeral merge.
-	if os.environ.get( 'BUILD_REASON', '' ) == 'PullRequest' :
-		commit = os.environ.get( 'SYSTEM_PULLREQUEST_SOURCECOMMITID' )
-	return commit
+commit = os.environ.get( 'BUILD_SOURCEVERSION' )
+# Azure merges the branch into its target in PR build, so
+# BUILD_SOURCEVERSION isn't correct as it references the ephemeral merge.
+if os.environ.get( 'BUILD_REASON', '' ) == 'PullRequest' :
+	commit = os.environ.get( 'SYSTEM_PULLREQUEST_SOURCECOMMITID' )
 
 formatVars = {
 	"buildTypeSuffix" : "-debug" if os.environ.get( "BUILD_TYPE", "" ) == "DEBUG" else "",
 	"platform" : "macos" if sys.platform == "darwin" else "linux",
 	"timestamp" : datetime.datetime.now().strftime( "%Y%m%d%H%M" ),
 	"pullRequest" : os.environ.get( "SYSTEM_PULLREQUEST_PULLREQUESTNUMBER", "UNKNOWN" ),
-	"commit" : getCommitHash()
+	"shortCommit" : commit[:8],
+	"branch" : os.environ.get( "BUILD_SOURCEBRANCHNAME", "unknown" ),
+	"jobName" : os.environ.get( "AGENT_JOBNAME", "unknown" ).lower()
 }
 
 nameFormats = {
-	"default" : "gaffer-{timestamp}-{commit}-{platform}{buildTypeSuffix}",
-	"PullRequest" : "gaffer-PR{pullRequest}-{commit}-{platform}{buildTypeSuffix}"
+	"default" : "gaffer-{timestamp}-{shortCommit}-{platform}{buildTypeSuffix}",
+	"PullRequest" : "gaffer-pr{pullRequest}-{shortCommit}-{platform}{buildTypeSuffix}",
+	"Schedule" : "gaffer-{jobName}-{timestamp}-{branch}-{shortCommit}-{platform}{buildTypeSuffix}"
```

Comment:
This seems overly verbose. As per my previous comment, let's remove branch because nightlies will just be for master. Let's remove jobName because I don't know what it is or why I would want it. And let's keep hours and minutes out of the timestamp because there's only one a night.

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/3248/commits/0125d9ff8321f0b11ce9f70ae0b9713b5db6cb4f